tests/console/commands/test_build.py

```diff
@@ -82,6 +83,35 @@ def test_build_with_local_version_label(
     assert all(archive.exists() for archive in build_artifacts)
 
 
+@pytest.mark.parametrize("clean", [True, False])
+def test_build_with_clean(
+    tmp_tester: CommandTester, tmp_project_path: Path, tmp_poetry: Poetry, clean: bool
+) -> None:
+    dist_dir = tmp_project_path.joinpath("dist")
+    dist_dir.joinpath("hello").touch(exist_ok=True)
+
+    tmp_tester.execute("--clean" if clean else "")
+    build_artifacts = tuple(dist_dir.glob("*"))
+
+    assert len(build_artifacts) == 2 if clean else 3
```

Comment:
Could we put the amount of artifacts and `--clean` param (line 93) in parametrize? I don't like the idea of conditionals in tests. It's a nitpick though, if you disagree, disregard it.

Reply:
In principle I agree. However since the test is what is creating the files, I think the code block needs to specify it rather than a parameter.  We can make it more clearer if you prefer? For example do one of he following.

1. Move the ternary conditional into an explicit block.
2. Make a count of the existing files in the directory and then assert it is `count + 2`.

But I suspect neither would be qualitatively better.